### DISCUSSION TITLE
IDE for SQL

### DISCUSSION THREAD

**Discussion by StillKill** [community/community#146255] [Programming Help] [answered]
### Body

I'm new to programming and currently learning SQL. Could experienced colleagues recommend an IDE for working with databases?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by NickOsipov** [comment, marked as answer]
@StillKill Hello!

I'm recommend DBeaver or VS Code with SQL Tools extension.

https://dbeaver.io/
https://marketplace.visualstudio.com/items?itemName=mtxr.sqltools

DBeaver offers:

1. A free version with extensive functionality
2. Support for multiple databases (PostgreSQL, MySQL, Oracle, and others)
3. Convenient visual table editor
4. Data viewing and editing capabilities
5. Code autocompletion
6. Built-in ERD (Entity Relationship Diagram) designer
7. Query execution with visual results
8. Database structure navigation

VSCode with SQL Tools extension provides:

1. Seamless integration with other development tools
2. SQL syntax highlighting
3. Code autocompletion
4. Query history
5. Ability to save frequently used queries
6. Multi-platform support
7. Lightweight and fast performance
8. Customizable interface

For a beginner, I would recommend starting with DBeaver as it's more intuitive and has all necessary tools available right "out of the box." The visual interface makes it easier to understand database structure and relationships. Once you become more comfortable with SQL, you might want to try VSCode, especially if you're already using it for other programming tasks. VSCode is particularly good if you work with multiple programming languages and want a single, unified development environment.
Both options are free and open-source, so you can try them both to see which one better suits your workflow.

**Reply by StillKill** [reply]
Hi, thanks a lot!
